pkg/controller/elasticache/user/setup.go

```diff
@@ -0,0 +1,200 @@
+	}
+	meta.SetExternalName(cr, awsclients.StringValue(obj.UserId))
+	return cre, nil
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.User, obj *svcsdk.DescribeUsersInput) error {
+	obj.UserId = awsclients.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.User, obj *svcsdk.DescribeUsersOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	switch awsclients.StringValue(obj.Users[0].Status) {
+	case "active":
+		cr.SetConditions(xpv1.Available())
+	case "modifying":
+		cr.SetConditions(xpv1.Unavailable())
+	case "deleting":
+		cr.SetConditions(xpv1.Deleting())
+	}
+
+	return obs, nil
+}
+
+func preDelete(_ context.Context, cr *svcapitypes.User, obj *svcsdk.DeleteUserInput) (bool, error) {
+	obj.UserId = awsclients.String(meta.GetExternalName(cr))
+	return false, nil
+}
+
+func postDelete(_ context.Context, cr *svcapitypes.User, obj *svcsdk.DeleteUserOutput, err error) error {
+	if err != nil {
+		if strings.Contains(err.Error(), svcsdk.ErrCodeInvalidUserStateFault) {
+			// skip: failed to delete User: InvalidUserState: User has status deleting.
+			return nil
+		}
+		return err
+	}
+	return err
+}
+
+func (e *hooks) isUpToDate(cr *svcapitypes.User, obj *svcsdk.DescribeUsersOutput) (bool, error) {
+	ctx := context.Background()
+
+	if awsclients.StringValue(obj.Users[0].Status) != active {
+		return true, nil
+	}
+
+	// to fields ex. $AccessString vs. $AccessString -@all
+	if awsclients.StringValue(cr.Spec.ForProvider.AccessString) != strings.Fields(awsclients.StringValue(obj.Users[0].AccessString))[0] {
+		return false, nil
+	}
+
+	isUpToDate, err := elasticache.PasswordsUpToDate(ctx, e.kube, cr.Spec.ForProvider.PasswordSecretRef, cr.Spec.WriteConnectionSecretToReference)
+	if err != nil {
+		return false, err
+	}
+	return isUpToDate, nil
+}
+
+func (e *hooks) preUpdate(ctx context.Context, cr *svcapitypes.User, obj *svcsdk.ModifyUserInput) error {
+	obj.UserId = awsclients.String(meta.GetExternalName(cr))
+
+	pwds, err := elasticache.GetPasswords(ctx, e.kube, cr.Spec.ForProvider.PasswordSecretRef)
+	if err != nil {
+		return err
+	}
+
+	obj.Passwords = pwds
+
+	return nil
+}
+
+func (e *hooks) postUpdate(ctx context.Context, cr *svcapitypes.User, obj *svcsdk.ModifyUserOutput, upd managed.ExternalUpdate, err error) (managed.ExternalUpdate, error) {
+	if err != nil {
+		return upd, err
+	}
+
+	pwds, err := elasticache.GetPasswords(ctx, e.kube, cr.Spec.ForProvider.PasswordSecretRef)
+	if err != nil {
+		return upd, err
+	}
+
+	connectionDetails := managed.ConnectionDetails{}
+
+	if len(pwds) > 0 {
+		connectionDetails[elasticache.PasswordKey1] = []byte(awsclients.StringValue(pwds[0]))
+		if len(pwds) > 1 {
+			connectionDetails[elasticache.PasswordKey2] = []byte(awsclients.StringValue(pwds[1]))
+		} else {
+			connectionDetails[elasticache.PasswordKey2] = nil
```

Comment:
I don't think this is a good approach. If we want to support more than one password we should write all passwords into the secret using a loop and `key := fmt.Sprintf("%s%d", prefix, i)`.

And in addition to that, we can set the first password to the default key `password`.

Reply:
only possible is 1 or 2 password 
if you remove the second password we remove it